### ISSUE TITLE
Local progress isn't synched to server anymore

### ISSUE THREAD

**Issue by JakobTischler** [advplyr/audiobookshelf-app#926]
### Steps to reproduce

1. Online and connected to server.
1. Start or resume a locally downloaded book.
2. Check history for sync status.
3. Red error icon; progress isn't synched at all.

### Expected behaviour
- The progress should be synched to the sever.

### Actual behaviour
- The progress is not synched to the server. When starting up the app, the book shortly appears in the "Continue Listening" section. Then, when fully connected to the server, the book disappears and the synched books are displayed there instead.
- 
![image](https://github.com/advplyr/audiobookshelf-app/assets/748857/4bd49058-6ab5-423b-90e9-e82c7ff36956)
- When I start / resume a streamed book, everything works as expected and its progress is synched.

I _think_ this started with .66b. I've tried removing the app's cache files; reconnecting to the server manually, to no avail.

### Environment data

Server version: v2.50, but also happend on v2.4x

Audiobookshelf Version:

- [x] Android App? v0.66 beta

#### Android Issue
* Android version: 14
* Device model: Google Pixel 7
* Stock or customized system: Stock

**Comment by advplyr** [comment]
If you go to the local media page for that download you should see the server address that it was downloaded on. Make sure that server address matches the one you are currently connected to.

**Comment by JakobTischler** [comment]
So the item's server address and the URI of the server I use are completely identical. The server hasn't changed ever since I started running ABS (other than updating versions). 

Based on that info I did try deleting and re-downloading the audiobook, though, and that did the trick. All listening progress was lost, though. I have around 10 or so more books and I have downloaded a long-ish time ago, so I assume they will have the same fate.

I think it might have had something to do with the switch from 2.3 to 2.4 and the conversion of the ABS data (?) ...
